runtime/near-vm-logic/src/logic.rs

```diff
@@ -1462,42 +1651,28 @@ impl<'a> VMLogic<'a> {
         value_ptr: u64,
         register_id: u64,
     ) -> Result<u64> {
-        let Self {
-            memory,
-            registers,
-            gas_counter,
-            config,
-            fees_config,
-            valid_iterators,
-            invalid_iterators,
-            ext,
-            ..
-        } = self;
-        gas_counter.pay_base(config.ext_costs.storage_write_base)?;
+        self.gas_counter.pay_base(base)?;
+        self.gas_counter.pay_base(storage_write_base)?;
         // All iterators that were valid now become invalid
-        for invalidated_iter_idx in valid_iterators.drain() {
-            ext.storage_iter_drop(invalidated_iter_idx)?;
-            invalid_iterators.insert(invalidated_iter_idx);
+        for invalidated_iter_idx in self.valid_iterators.drain() {
+            self.ext.storage_iter_drop(invalidated_iter_idx)?;
+            self.invalid_iterators.insert(invalidated_iter_idx);
         }
-        let key = Self::get_from_memory_or_register(*memory, registers, key_ptr, key_len)?;
-        let value = Self::get_from_memory_or_register(*memory, registers, value_ptr, value_len)?;
-        gas_counter.pay_per_byte(config.ext_costs.storage_write_value_byte, key.len() as u64)?;
-        gas_counter.pay_per_byte(config.ext_costs.storage_write_value_byte, value.len() as u64)?;
+        let key = self.get_vec_from_memory_or_register(key_ptr, key_len)?;
+        let value = self.get_vec_from_memory_or_register(value_ptr, value_len)?;
+        self.gas_counter.pay_per_byte(storage_write_key_byte, key.len() as u64)?;
+        self.gas_counter.pay_per_byte(storage_write_value_byte, value.len() as u64)?;
         let evicted = self.ext.storage_set(&key, &value)?;
-        let storage_config = &fees_config.storage_usage_config;
+        let storage_config = &self.fees_config.storage_usage_config;
         match evicted {
             Some(old_value) => {
                 self.current_storage_usage -=
```

Comment:
Might panic with underflow for incorrect `current_storage_usage` in account during proof

Reply:
FYI @fckt